### PR TITLE
[LA 274] - Streamline Redshift external schema reset and table creation process. 

### PR DESCRIPTION
**Pull request by sandeepmjay**
Since we are working with External schema and table structures some of the traditional clauses like IF EXISTS and IF NOT EXISTS are not widely supported in Spectrum. So for DB initialization and resets we query the SVV_EXTERNAL_TABLES where all the external schema definition are stored on Redshift cluster to identify the active schemas and tables.

once identified these are reset by Dropping all the schemas and tables. Fresh table creation scripts are run pointing to the new data location at the time of the refresh
https://jira.ets.berkeley.edu/jira/browse/LA-274

Also, added logic for repointing DB tables, fixed some minor error handling and logging.  Cleaned up SQL template generation logic.

### CODE REVIEW

File: lib/sync/db-templates/dbCreation.template.sql [hunk 1]
```diff
@@ -499,6 +463,76 @@ FIELDS TERMINATED BY '\t'
...
+  discussion_id BIGINT,
+  conversation_id BIGINT,
+  assignment_id BIGINT,
+  url VARCHAR,
+  user_agent VARCHAR,
+  http_method VARCHAR,
+  remote_ip VARCHAR,
+  interaction_micros BIGINT,
+  web_application_controller VARCHAR,
+  web_applicaiton_action VARCHAR,
```
Comment (paulkerschen) [inline lib/sync/db-templates/dbCreation.template.sql]: @sandeepmjay typo `applicaiton`

File: lib/sync/db-templates/dbCreation.template.sql [hunk 2]
```diff
@@ -499,6 +463,76 @@ FIELDS TERMINATED BY '\t'
...
```
Comment (paulkerschen) [inline lib/sync/db-templates/dbCreation.template.sql]: @sandeepmjay Same as above

File: lib/sync/db.js [hunk 3]
```diff
@@ -216,6 +286,31 @@ var createDatabase = function(callback) {
...
+    }
+
+    repointDatabase(function(err) {
+      if (err) {
+        log.error({err: err.message}, 'Canvas Data repoint on Redshift instance failed');
+
+        return callback(err);
+      }
+
+      log.info('Canvas external schemas on Redshift instance has been repointed to new daily location.');
```
Comment (paulkerschen) [inline lib/sync/db.js]: @sandeepmjay One schema or many? Either "schemas... have been repointed" or "schema... has been repointed"

**Comment by paulkerschen** [comment]
Thanks, @sandeepmjay, looks good to merge as soon as a couple of typos are fixed.

**Comment by sandeepmjay** [comment]
Thanks @paulkerschen. Ready for merge.

**Pull request merged**